pkg/kube/krt/collection.go

```diff
@@ -343,6 +343,8 @@ func (h *manyCollection[I, O]) onPrimaryInputEvent(items []Event[I]) {
 func (h *manyCollection[I, O]) onPrimaryInputEventLocked(items []Event[I]) {
 	var events []Event[O]
 	recomputedResults := make([]map[Key[O]]O, len(items))
+
+	pendingDepStateUpdates := make(map[Key[I]]*collectionDependencyTracker[I, O], len(items))
```

Comment:
the collection fix need a test

Reply:
it's tested by the test added in the PR - that is how I found it

Reply:
The one in the workload_test.go?

Reply:
yes